store/day_store.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package store
+
+import (
+	"time"
+)
+
+// DayStore holds an 1-hour StatStore whose average is taken every one hour.
+// DayStore also holds a 1-day StatStore where the individual sub-averages are placed.
+type DayStore struct {
+	Hour          StatStore
+	Day           StatStore
+	DayMaxima     TimeStore
+	created       time.Time
+	lastUpdate    time.Time
+	dayResolution time.Duration
+}
+
+// DayPassed returns true iff the DayStore is populated with more than 24 hours of data.
+func (ds *DayStore) DayPassed() bool {
+	lastTP := ds.Hour.Last()
+	if lastTP == nil || !ds.created.Add(24*time.Hour).Before(lastTP.Timestamp) {
+		return false
+	}
+	return true
```

Comment:
This is actually a question - do you generally accept `if cond { return false } return true` instead of `return cond`? I know `cond` in this case is rather large, but still wondering.